src/cli.rs

```diff
@@ -341,7 +341,15 @@ impl Config {
                 secret,
             })
         } else {
-            None
+            let conflicts = args.reward_address.is_some() || args.secret.is_some();
+
+            if conflicts {
+                return Err(AppError::Server(
+                    "Cannot set --reward-address or --secret in solitary mode (--solitary)".into(),
```

Comment:
Nice, so you found a _sort of_ a bug!

Reply:
@michalrus Yeah, let's call it a small improvement ^^